frontend/src/metabase/static-viz/components/LineAreaBarChart/LineAreaBarChart.tsx

```diff
@@ -51,25 +44,7 @@ const LineAreaBarChart = ({
     goalColor: getColor("text-medium"),
   };
 
-  const minTickSize = chartStyle.axes.ticks.fontSize * 1.5;
-  const xValuesCount = getXValuesCount(series);
-  const chartSize = calculateChartSize(settings, xValuesCount, minTickSize);
-  const adjustedSettings = adjustSettings(
```

Comment:
I changed `adjustSettings` into `getXAxisProps` which returns props for `<AxisBottom />` instead. Since there are certain logics inside `adjustSettings` that I want to keep. And I don't want to have to pass call this function twice I move it inside `<XYchart />` instead

Reply:
My idea was to make `XYChart` declarative as much as possible so that it does not change its visual settings based on the data provided. One of the reasons why our non-static charts is hardly maintainable is because we have a single render function where data is being transformed and based on it and visual settings it computes actual visual settings. But if we separate computing actual settings and just rendering according to computed spec, it will be a lot easier and testable. What do you think about it?

Reply:
Totally agreed, I was carried away by the refactoring and try to not recompute everything. Anyways, almost all of the PR should be reverted as per our pod sync yesterday to keep the code simple and focus on shipping more low-hanging fruits.